spec/acceptance/agent_and_server_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'spec_helper_acceptance'
+require 'simp/beaker_helpers'
+
+include Simp::BeakerHelpers
+
+describe 'hyperglass::server class' do
+  # https://github.com/simp/rubygem-simp-beaker-helpers#run_fake_pki_ca_on
+  # rubocop:disable RSpec/EmptyExampleGroup
+  context 'it setups a fake CA' do
+    Dir.mktmpdir do |cert_dir|
+      run_fake_pki_ca_on(hosts.first, hosts, cert_dir)
+      hosts.each { |sut| copy_pki_to(sut, cert_dir, '/etc/puppetlabs/puppet/ssl/') }
+    end
+    shell('mkdir -p /etc/puppetlabs/puppet/ssl/certs/; mv /etc/puppetlabs/puppet/ssl/pki/cacerts/cacerts.pem /etc/puppetlabs/puppet/ssl/certs/ca.pem')
+    shell('mkdir -p /etc/puppetlabs/puppet/ssl/private_keys; mv /etc/puppetlabs/puppet/ssl/pki/private/centos7-64-1.pem /etc/puppetlabs/puppet/ssl/private_keys/')
+    shell('mv /etc/puppetlabs/puppet/ssl/pki/public/centos7-64-1.pub /etc/puppetlabs/puppet/ssl/certs/centos7-64-1.pem')
+  end
+  context 'default parameters' do
+    # Using puppet_apply as a helper
+    it 'works with no errors' do
+      pp = 'include hyperglass::server; include hyperglass::agent'
+
+      # Run it twice and test for idempotency
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_failures: true) # redis dependency needs two runs to start properly on selinux nodes
```

Comment:
A workaround for that is to install the Redis package in the acceptance helper: https://github.com/theforeman/puppet-pulpcore/blob/3de6b648818410929fb9e1be7645f39063f9c61a/spec/setup_acceptance_node.pp#L13-L16 does that. You can refer to https://tickets.puppetlabs.com/browse/PUP-10548 as well.

Reply:
I'm not sure if this is a sane approach. The module can install redis. I think to not influence the resource ordering it should come from the module. It added an if clause that checks if selinux is enabled and only does the second `apply_manifest()` in those environments. What do you think about that?